src/com/amazon/ionelement/impl/SeqElementBase.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.ionelement.impl
+
+import com.amazon.ion.IonWriter
+import com.amazon.ionelement.api.AnyElement
+import com.amazon.ionelement.api.MetaContainer
+import com.amazon.ionelement.api.SeqElement
+import com.amazon.ionelement.api.emptyMetaContainer
+
+internal abstract class SeqElementBase(
+    override val values: List<AnyElement>,
+    override val annotations: List<String> = emptyList(),
+    override val metas: MetaContainer = emptyMetaContainer()
+): IonElementBase(), SeqElement {
```

Comment:
By extending `IonElementBase`, all the accessors defined in `AnyElement` are accessible in `SeqElementBase`. I understand that this is internal class so there's no impact on user side, just curious if there's a specific reason that `IonElementBase` is required?

Reply:
So, it shouldn't be confusing as long as the user doesn't use the `IonElementBase` type directly in their code.

`IonElementBase` provides the majority of the implementation of `AnyElement` (which probably means it should be renamed to `AnyElementBase`).

Reply:
Yeah I'm aware that the implementation is hidden from customer. Whether it extend `IonElementBase()` or not, user won't notice any difference. Just curious why `SeqElementBase` need to support methods defined in `AnyElement`? If we remove `IonElementBase()` here, what would break?